arch/x86/core/acpi.c

```diff
@@ -27,21 +28,34 @@ static void find_rsdp(void)
 	uint8_t *bda_seg, *zero_page_base;
 	uint64_t *search;
 	uintptr_t search_phys, rsdp_phys = 0U;
-	size_t search_length, rsdp_length;
+	size_t search_length = 0U, rsdp_length;
 
 	if (is_rsdp_searched) {
 		/* Looking up for RSDP has already been done */
 		return;
 	}
 
+	/* Let's first get it from EFI, if enabled */
```

Comment:
Does this break test coverage for the legacy ACPI methods?  I'm pretty sure all our hardware platforms are EFI, and qemu doesn't do ACPI as we set it up, right?

Reply:
Qemu can provide a basic support of ACPI, see tests/arch/x86/info (both qemu_x86 and qemu_x86_64). So as long as CONFIG_X86_EFI is not enabled on qemu (it can be, but it's a bit of setup work, see https://docs.zephyrproject.org/latest/boards/x86/qemu_x86/doc/index.html#programming-and-debugging ), the legacy ACPI will be covered.